docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
+3. `ListCommand` loops through every `Item` in `itemList` and prints them line by line
+   and numbers them.
+
+### Storage
+
+#### Initialisation
+
+The following diagram shows the sequence diagram illustrating how `Storage` is initialised when the program first launches.
+
+![StorageInitialisationSequenceDiagram](img/StorageInitialisationSequenceDiagram.png)
+
+1. `InvMgr` calls the `Storage(filePath)` constructor to create a `Storage` object. `filePath` is a `String` indicating where the data file to be loaded is found.
+2. `Storage(filePath)` will check if the file at `filePath` exists. If it does, it returns a `Path` object pointing to the data file.
+3. If not, the relevant files and subdirectories are created before returning the corresponding `Path` object.
+4. The new `Storage` object will have its `dataPath` attribute set to the `Path` object earlier, and its `filePath` attribute set to the `filePath` passed into the constructor.
+
+#### Loading data
+
+The following diagram shows the sequence diagram illustrating how the data file is loaded. Typically, this is only run once when the program first launches.
+
+![StorageLoadSequenceDiagram](img/StorageLoadSequenceDiagram.png)
+
+1. `InvMgr` calls the `load()` method of `storage`.
+2. `storage` initialises `Gson()` as `gson`, a library used to serialize and deserialize JSON objects into their relevant Java objects.
+3. `storage` will then load the contents of the file at `dataPath` into the `wholeJsonData` `String`. The exact details are not shown in the diagram.
+4. `storage` then calls the `fromJson(wholeJsonData)` method of `gson`.
+5. An `ArrayList<Item>` may be returned by `fromJson()` method. If it is not, a new empty list is created.
+6. `storage` returns `ArrayList<Item>` to `InvMgr`. This will be used to create the `ItemList`, but will not be shown here.
+
+#### Saving data
+
+The following diagram shows the sequence diagram illustrating how the data file is saved. Typically, this is done after each `Command` is run.
+
+![StorageSaveSequenceDiagram](img/StorageSaveSequenceDiagram.png)
+
+1. `InvMgr` calls the `save(itemList)` method of `storage`.
+2. `storage` initialises `Gson()` as `gson`, a library used to serialize and deserialize JSON objects into their relevant Java objects.
+3. `storage` calls the `toJson(itemList)` method of `gson`. This returns a `String` named `serializedItems` which contains the JSON String representing `itemList`.
+4. `storage` then writes `serializedItems` to the file at `dataFile`.
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+CCA Clubs that require assistance in inventory management, especially for fast typists who prefer CLI over GUI.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+1. Centralised management of resources that ensures accurate and timely allocation of equipment to students
+2. Increases the ease and efficiency of resource management
+3. More organised
```

Comment:
Consider elaborating on this value proposition as it is quite vague. Could possible extend to be similiar like point 1.